xlive/Blam/Engine/interface/user_interface_player_profile_list.cpp

```diff
@@ -0,0 +1,156 @@
+
+void c_player_profile_list::on_screen_leave()
+{
+	INVOKE_TYPE(0x23685F, 0x0, void(__thiscall*)(c_player_profile_list*), this);
+}
+
+void c_player_profile_list::update()
+{
+	INVOKE_TYPE(0x2375A4, 0x0, void(__thiscall*)(c_player_profile_list*), this);
+}
+
+void c_player_profile_list::update_list_items(c_list_item_widget* item, int32 skin_index)
+{
+	INVOKE_TYPE(0x23686E, 0x0, void(__thiscall*)(c_player_profile_list*, c_list_item_widget*, int32), this, item, skin_index);
+}
+
+void c_player_profile_list::unknown_function_2(int32 a1)
+{
+	INVOKE_TYPE(0x236B1E, 0x0, void(__thiscall*)(c_player_profile_list*, int32), this, a1);
+}
+
+__declspec(naked) void jmp_c_player_profile_list__update_displayed_profiles() { __asm { jmp c_player_profile_list::update_displayed_profiles } };
+
+void c_player_profile_list::update_displayed_profiles()
+{
+	uint32 profile_storage_size;
+	s_user_interface_player_profile_storage* profile_storage = this->get_displayed_player_profile_storage(&profile_storage_size);
+
+	uint32 profile_indices[16];
+	int32 current_child_widget_index = 0;
+
+	int32 enumerated_file_index = NONE;
+
+	c_list_item_widget* current_child = (c_list_item_widget*)this->m_child_widget;
+
+	memset(profile_indices, NONE, sizeof(uint32) * 16);
+	while(current_child && current_child_widget_index < 16)
+	{
+		if (current_child_widget_index >= 16)
+			break;
+
+		if(current_child->get_last_data_index() != NONE)
+		{
+			void* item = datum_get(this->m_list_data, current_child->get_last_data_index());
+			if (item)
+				profile_indices[current_child_widget_index] = *((int32*)item + 1);
+		}
+		current_child = (c_list_item_widget*)current_child->next_widget;
+		++current_child_widget_index;
+	}
+
+	if(profile_storage_size > 0)
+	{
+		uint32 current_profile_storage_index = 0;
+		
+		do
+		{
+			enumerated_file_index = profile_storage[current_profile_storage_index].enumerated_file_index;
+
+			int32 current_player_profile_indices_index = 0;
+			while(true)
+			{
+				if(enumerated_file_index == profile_indices[current_player_profile_indices_index])
+				{
+					if (enumerated_file_index != user_interface_globals_get_edit_player_profile_index())
+					{
+						profile_indices[current_player_profile_indices_index] = NONE;
+						break;
+					}
+				}
+
+				if (++current_player_profile_indices_index >= 16)
+					break;
+			}
+			if(current_player_profile_indices_index != 16)
+				profile_indices[current_player_profile_indices_index] = NONE;
+
+			if (current_player_profile_indices_index == 16 && enumerated_file_index != NONE)
+				profile_storage[current_profile_storage_index].enumerated_file_index = NONE;
+
+			++current_profile_storage_index;
+		} while (current_profile_storage_index < profile_storage_size);
+	}
+
+	for(int32 index = 0; index < 16; ++index)
+	{
+		const uint32 current_profile_index = profile_indices[index];
+		if(current_profile_index != NONE)
+		{
+			if(profile_storage_size > 0)
+			{
+				uint32 current_profile_storage_index = 0;
+				do
```

Comment:
same here, for loop
de-compilation might represent this as a do while, since the code is optimized automatically by the compiler